bundle/artifacts/upload.go

```diff
@@ -41,6 +43,32 @@ func (m *upload) Apply(ctx context.Context, b *bundle.Bundle) error {
 		return fmt.Errorf("artifact source is not configured: %s", m.name)
 	}
 
+	// Check if source paths are absolute, if not, make them absolute
+	for k := range artifact.Files {
+		f := &artifact.Files[k]
+		if !filepath.IsAbs(f.Source) {
+			f.Source = filepath.Join(b.Config.Path, f.Source)
+		}
+	}
+
+	// Expand any glob reference in files source path
+	files := make([]config.ArtifactFile, 0, len(artifact.Files))
+	for _, f := range artifact.Files {
+		matches, err := filepath.Glob(f.Source)
+		// If the source is not a glob pattern or no matches, just add it to the list
+		if err != nil || len(matches) == 0 {
```

Comment:
If it is a glob and doesn't match anything, this should be an error.

The errors that show if we try to use `dist/*.whl` if it doesn't exist will likely be cryptic otherwise.

Reply:
There is no built-in function to check if a string is a pattern, but we could DIY something easy by checking if the string includes characters commonly used in these patterns (`*`, `?`, `[`, `]`, etc).

Reply:
I don't think we even need to check if it's glob or not. Even if it's a full concrete path and file does not exist we can fail with an error